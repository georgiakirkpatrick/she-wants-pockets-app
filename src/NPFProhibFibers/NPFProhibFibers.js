import React from 'react'
import formData from '../FORM_DATA'
import FormPage from '../FormPage/FormPage'
import FormCheckboxSection from '../FormCheckboxSection/FormCheckboxSection'
import FormPromptWithSub from '../FormPromptWithSub/FormPromptWithSub'
import NPFFooter from '../NPFFooter/NPFFooter'

const noneOption = [
    {
        text: 'None of these',
        id: 100,
        checked: false
    }
]

const NPFProhibFibers = props => {
    const nextButton = () => {
        const prohibFibs = []

        Object.keys(props.pFiberChecks).forEach(key => {
            if (props.pFiberChecks[key]) {
                prohibFibs.push(key)
            }
        })

        if (prohibFibs.length === 0 && !props.none[100]) {
            alert('Please select an option.')
        } else {
            props.setPage(props.currentPage + props.pageTurns)
        }
    }

    return (
        <div>
            <FormPage title='Fibers'>
                <FormPromptWithSub
                    prompt='Does the product contain any of following materials?'
                    promptSubtitle="Check all that apply and click 'next'."
                />

                <FormCheckboxSection 
                    options={formData.prohibitedFibers}
                    selectedOptions={props.pFiberChecks}
                    handleChange={props.pFiberChange}
                />

                <FormCheckboxSection  
                    options={noneOption}
                    selectedOptions={props.none}
                    handleChange={props.noneChange}
                />
            </FormPage>

            <NPFFooter
                buttons='prevNext'
                previousButton={() => props.setPage(props.currentPage - props.pageTurns)} 
                nextButton={() => nextButton()}
            />
        </div>
    )
}

NPFProhibFibers.defaultProps = {
    currentPage: 1,
    handleChange: () => {},
    none: {},
    noneChange: () => {},
    pageTurns: 1,
    pFiberChange: () => {},
    pFiberChecks: {},
    selectedOptions: [],
    setPage: () => {},
    setPFiberChecks: () => {}
}

export default NPFProhibFibers