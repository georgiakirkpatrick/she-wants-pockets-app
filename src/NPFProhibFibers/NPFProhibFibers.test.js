import React from 'react'
import ReactDOM from 'react-dom'
import NPFProhibFibers from './NPFProhibFibers'
import renderer from 'react-test-renderer'

describe ('NPFProhibFibers', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<NPFProhibFibers />, div
        )
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<NPFProhibFibers />)
            .toJSON()
        expect(tree).toMatchSnapshot()    
    })
})