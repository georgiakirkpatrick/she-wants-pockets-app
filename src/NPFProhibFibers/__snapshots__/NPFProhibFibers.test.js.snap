// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NPFProhibFibers renders the UI as expected 1`] = `
<div>
  <section
    className="FormPage"
  >
    <h1>
      Fibers
    </h1>
    <form
      id="form-page"
    >
      <div
        className="FormPromptWithSub"
      >
        <p
          className="FormPromptWithSub__prompt"
        >
          Does the product contain any of following materials?
        </p>
        <p
          className="FormPromptWithSub__subtitle"
        >
          Check all that apply and click 'next'.
        </p>
      </div>
      <div
        className="FormCheckboxSection"
      >
        <div
          className="empty"
        />
        <div
          className="FormCheckboxSection__options-container"
        >
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id="acetate"
              name="acetate"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor="acetate"
            >
              Acetate
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id="acrylic"
              name="acrylic"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor="acrylic"
            >
              Acrylic
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id="microfiber"
              name="microfiber"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor="microfiber"
            >
              Microfiber
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id="neoprene"
              name="neoprene"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor="neoprene"
            >
              Neoprene
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id="nylon"
              name="nylon"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor="nylon"
            >
              Nylon
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id="olefin"
              name="olefin"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor="olefin"
            >
              Olefin
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id="polyamide"
              name="polyamide"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor="polyamide"
            >
              Polyamide
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id="polyester"
              name="polyester"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor="polyester"
            >
              Polyester
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id="polyurethane"
              name="polyurethane"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor="polyurethane"
            >
              Polyurethane
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id="vinyl"
              name="vinyl"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor="vinyl"
            >
              Vinyl
            </label>
          </div>
          <div
            className="FormCheckboxSection__option"
          >
            <input
              id="none"
              name="none"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor="none"
            >
              None of these
            </label>
          </div>
        </div>
      </div>
    </form>
    <div />
  </section>
  <footer
    className="NPFFooter"
  >
    <button
      className="NPFFooter_previous"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-arrow-left fa-w-14 NPFFooter_arrow"
        data-icon="arrow-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        PREVIOUS
      </span>
    </button>
    <button
      className="NPFFooter_next"
      onClick={[Function]}
      type="button"
    >
      <span>
        NEXT
      </span>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-arrow-right fa-w-14 NPFFooter_arrow"
        data-icon="arrow-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </footer>
</div>
`;
